aws_s3bucket acceptance tests failing

All S3BucketObject tests pass, but all S3Bucket_* tests fail.
I see the tf-test-buckets-XXXX buckets getting created and removed on my S3 account, so I'm not sure why these are failing.
➜  terraform git:(master) ✗ make testacc TEST=./builtin/providers/aws TESTARGS='-run=AWSS3Bucket'
go generate ./...
TF_ACC=1 go test ./builtin/providers/aws -v -run=AWSS3Bucket -timeout 90m
=== RUN   TestAccAWSS3BucketObject_source
--- PASS: TestAccAWSS3BucketObject_source (11.02s)
=== RUN   TestAccAWSS3BucketObject_content
--- PASS: TestAccAWSS3BucketObject_content (12.36s)
=== RUN   TestAccAWSS3BucketObject_withContentCharacteristics
--- PASS: TestAccAWSS3BucketObject_withContentCharacteristics (12.66s)
=== RUN   TestAccAWSS3Bucket_basic
--- FAIL: TestAccAWSS3Bucket_basic (10.99s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_Policy
--- FAIL: TestAccAWSS3Bucket_Policy (20.66s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_UpdateAcl
--- FAIL: TestAccAWSS3Bucket_UpdateAcl (18.24s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_Website_Simple
--- FAIL: TestAccAWSS3Bucket_Website_Simple (25.35s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_WebsiteRedirect
--- FAIL: TestAccAWSS3Bucket_WebsiteRedirect (20.42s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_shouldFailNotFound
--- FAIL: TestAccAWSS3Bucket_shouldFailNotFound (6.43s)
	testing.go:144: Step 0 error: After applying this step and refreshing, the plan was not empty:

		DIFF:

		CREATE: aws_s3_bucket.bucket
		  acl:              "" => "public-read"
		  arn:              "" => "<computed>"
		  bucket:           "" => "tf-test-bucket-2003001439837752215" (forces new resource)
		  force_destroy:    "" => "0"
		  hosted_zone_id:   "" => "<computed>"
		  region:           "" => "<computed>"
		  website_domain:   "" => "<computed>"
		  website_endpoint: "" => "<computed>"

		STATE:

		<no state>
=== RUN   TestAccAWSS3Bucket_Versioning
--- FAIL: TestAccAWSS3Bucket_Versioning (25.21s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
=== RUN   TestAccAWSS3Bucket_Cors
--- FAIL: TestAccAWSS3Bucket_Cors (10.61s)
	testing.go:165: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id:

		State: <no state>
FAIL

--- a/builtin/providers/aws/resource_aws_s3_bucket_test.go
+++ b/builtin/providers/aws/resource_aws_s3_bucket_test.go
@@ -14,6 +14,7 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/s3"
 )
 
@@ -265,6 +266,9 @@ func testAccCheckAWSS3BucketDestroy(s *terraform.State) error {
 			Bucket: aws.String(rs.Primary.ID),
 		})
 		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NoSuchBucket" {
+				return nil
+			}
 			return err
 		}
 	}

